src/librustc_driver/driver.rs

```diff
@@ -667,7 +667,9 @@ pub fn phase_2_configure_and_expand<F>(sess: &Session,
                     new_path.push(path);
                 }
             }
-            env::set_var("PATH", &env::join_paths(new_path).unwrap());
+            if let Ok(p) = env::join_paths(new_path) {
```

Comment:
I think I missed something then. In here, we just join 2 paths, if one is invalid, we should only keep the other one, right? I'll try to update in this way then.